rsocket-core/src/main/java/io/rsocket/metadata/TracingMetadataCodec.java

```diff
@@ -0,0 +1,166 @@
+package io.rsocket.metadata;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufAllocator;
+
+public class TracingMetadataCodec {
+
+  static final int FLAG_EXTENDED_TRACE_ID_SIZE = 0b0000_1000;
+  static final int FLAG_INCLUDE_PARENT_ID = 0b0000_0100;
+  static final int FLAG_NOT_SAMPLED = 0b0001_0000;
+  static final int FLAG_SAMPLED = 0b0010_0000;
+  static final int FLAG_DEBUG = 0b0100_0000;
+  static final int FLAG_IDS_SET = 0b1000_0000;
+
+  public static ByteBuf encodeEmpty(ByteBufAllocator allocator, Flags flag) {
```

Comment:
Likewise some minimal Javadoc with an `@since 1.0` tag. Maybe a bit on the encode and decode methods but it doesn't need to hold up 1.0.